eq-author-api/utils/datastore.js

```diff
@@ -140,6 +140,40 @@ const addEventToHistory = (questionnaireId, historyEvent) =>
     );
   });
 
+const updateEventHistory = (questionnaireId, historyEventId, bodyText) =>
+  new Promise((resolve, reject) => {
+    QuestionnaireModel.queryOne({ id: { eq: questionnaireId } }).exec(
+      async (err, questionnaire) => {
+        if (err) {
+          reject(err);
+        }
+        const noteToEdit = find(
+          questionnaire.history,
+          history => history.id === historyEventId
+        );
+        noteToEdit.bodyText = bodyText;
+
+        await saveModel(new QuestionnaireModel(questionnaire));
+        resolve(questionnaire.history);
+      }
+    );
+  });
+
+const removeEventFromHistory = (questionnaireId, eventId) =>
+  new Promise((resolve, reject) => {
+    QuestionnaireModel.queryOne({ id: { eq: questionnaireId } }).exec(
+      async (err, questionnaire) => {
+        if (err) {
+          reject(err);
+        }
+        remove(questionnaire.history, history => history.id === eventId);
+
+        await saveModel(new QuestionnaireModel(questionnaire));
+        resolve(questionnaire.history);
+      }
+    );
+  });
```

Comment:
If you mutate the history objects inside the resolvers then pass that to the save model we should be able to delete both these functions.